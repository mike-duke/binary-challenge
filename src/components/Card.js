import React from 'react';

export const Card = ({article}) => {
  const { urlToImage, title, author, url } = article
  return (
    <article className="card" onClick={() => window.open(url)}>
      <div className="image-container" style={{backgroundImage: `URL(${urlToImage})`}}>
      </div>
      <h3>{title}</h3>
      <p>By {author}</p>
    </article>
  )
}