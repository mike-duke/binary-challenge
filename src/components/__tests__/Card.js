import React from 'react';
import { shallow } from 'enzyme';
import { Card } from '../Card';

describe('Card', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Card article={{title: 'Cats are cool', author: 'All the cats', source: {name: 'Cat News'}}} />);

    expect(wrapper).toMatchSnapshot();
  });
});