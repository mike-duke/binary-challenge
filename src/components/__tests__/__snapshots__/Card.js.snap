// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    article={
      Object {
        "author": "All the cats",
        "source": Object {
          "name": "Cat News",
        },
        "title": "Cats are cool",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="image-container"
          style={
            Object {
              "backgroundImage": "URL(undefined)",
            }
          }
        />,
        <h3>
          Cats are cool
        </h3>,
        <p>
          By 
          All the cats
        </p>,
        <p>
          For 
          <span>
            Cat News
          </span>
        </p>,
      ],
      "className": "card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "image-container",
          "style": Object {
            "backgroundImage": "URL(undefined)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cats are cool",
        },
        "ref": null,
        "rendered": "Cats are cool",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "By ",
            "All the cats",
          ],
        },
        "ref": null,
        "rendered": Array [
          "By ",
          "All the cats",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "For ",
            <span>
              Cat News
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "For ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cat News",
            },
            "ref": null,
            "rendered": "Cat News",
            "type": "span",
          },
        ],
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="image-container"
            style={
              Object {
                "backgroundImage": "URL(undefined)",
              }
            }
          />,
          <h3>
            Cats are cool
          </h3>,
          <p>
            By 
            All the cats
          </p>,
          <p>
            For 
            <span>
              Cat News
            </span>
          </p>,
        ],
        "className": "card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "image-container",
            "style": Object {
              "backgroundImage": "URL(undefined)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cats are cool",
          },
          "ref": null,
          "rendered": "Cats are cool",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "By ",
              "All the cats",
            ],
          },
          "ref": null,
          "rendered": Array [
            "By ",
            "All the cats",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "For ",
              <span>
                Cat News
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "For ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cat News",
              },
              "ref": null,
              "rendered": "Cat News",
              "type": "span",
            },
          ],
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
